pkg/build/api/validation/validation.go

```diff
@@ -23,11 +34,20 @@ func ValidateBuildConfig(config *buildapi.BuildConfig) errs.ValidationErrorList
 	allErrs := errs.ValidationErrorList{}
 	if len(config.Name) == 0 {
 		allErrs = append(allErrs, errs.NewFieldRequired("name", config.Name))
+	} else if !util.IsDNSSubdomain(config.Name) {
+		allErrs = append(allErrs, errs.NewFieldInvalid("name", config.Name, "name must be a valid subdomain"))
```

Comment:
Because kubernetes? https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/api-conventions.md https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/design/identifiers.md

----- Original Message -----
> > @@ -23,11 +34,20 @@ func ValidateBuildConfig(config *buildapi.BuildConfig)
> > errs.ValidationErrorList
> >  	allErrs := errs.ValidationErrorList{}
> >  	if len(config.Name) == 0 {
> >  		allErrs = append(allErrs, errs.NewFieldRequired("name", config.Name))
> > +	} else if !util.IsDNSSubdomain(config.Name) {
> > +		allErrs = append(allErrs, errs.NewFieldInvalid("name", config.Name,
> > "name must be a valid subdomain"))
> 
> why?
> 
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/openshift/origin/pull/679/files#r23217139